packages/manager/modules/pci/src/projects/new/service.js

```diff
@@ -0,0 +1,254 @@
+import find from 'lodash/find';
+import head from 'lodash/head';
+import map from 'lodash/map';
+
+import PciCartProject from './classes/cart.class';
+
+import { PCI_PROJECT_ORDER_CART } from './constants';
+
+export default class PciProjectNewService {
+  /* @ngInject */
+  constructor(orderCart, OvhApiCloud, OvhApiOrder) {
+    this.orderCart = orderCart;
+    this.OvhApiCloud = OvhApiCloud;
+    this.OvhApiOrder = OvhApiOrder;
+  }
+
+  checkEligibility(voucher = null) {
+    return this.OvhApiCloud.v6().getEligibility({ voucher }).$promise;
+  }
+
+  /**
+   *  Get order cart from it's id or get a new one
+   *  @param  {string} ovhSubsidiary The subsidiary of the possible new cart
+   *  @param  {string} [cartId]      A previously created cart id
+   *  @return {Promise}              That returns a cart.
+   */
+  async getOrderCart(ovhSubsidiary, cartId = null) {
```

Comment:
This should be done with a `$q` instead,  If not every caller will probably have to `$timeout` to retrigger digests

Reply:
It depends of the implementation. Here it is used by ui-router resolve. I think it is to the controller which call the method to handle the use of `$q` or not. What do you think?